packages/api/es/indexer.js

```diff
@@ -344,30 +387,39 @@ async function reingest(g) {
  * @returns {Promise} object with response from the three indexer
  */
 async function handlePayload(event) {
-  let payload;
-  const source = get(event, 'EventSource');
+  const executionModel = new Execution();
+  const granuleModel = new Granule();
+  const pdrModel = new Pdr();
 
-  if (source === 'aws:sns') {
-    payload = get(event, 'Sns.Message');
-    payload = JSON.parse(payload);
-  }
-  else {
-    payload = event;
-  }
+  const payload = event.EventSource === 'aws:sns'
+    ? JSON.parse(event.Sns.Message)
+    : event;
+
+  const executionRecord = await buildExecutionRecord(executionModel, payload);
+  const pdrRecord = buildPdrRecord(pdrModel, payload);
+  const granuleRecords = await buildGranuleRecords(granuleModel, payload);
 
-  let executionPromise;
-  const e = new Execution();
-  if (['failed', 'completed'].includes(payload.meta.status)) {
-    executionPromise = e.updateExecutionFromSns(payload);
+  const recordsCount = flatten([executionRecord, granuleRecords, pdrRecord]).length;
+
+  if (recordsCount > 10 || inTestMode()) {
+    if (executionRecord) await executionModel.create(executionRecord);
+    if (pdrRecord) await pdrModel.create(pdrRecord);
+    await granuleModel.create(granuleRecords);
   }
-  else {
-    executionPromise = e.createExecutionFromSns(payload);
+  else if (recordsCount > 0) {
+    const TransactItems = flatten([
+      buildTransactPut(executionModel.tableName, executionRecord),
+      buildTransactPut(pdrModel.tableName, pdrRecord),
+      granuleRecords.map(buildTransactPut(granuleModel.tableName))
```

Comment:
Does the transaction for granuleRecords include writing to the 'files' table?

Reply:
No, the files table is currently written to by a different lambda function, the [db-indexer](https://github.com/nasa/cumulus/blob/8b5252fc13d83670143510bfaf0ee1bb361f07f9/packages/api/lambdas/db-indexer.js#L40).